userspace/graph-store/sur-file-walkthrough.md

```diff
@@ -0,0 +1,336 @@
++++
+title = "`sur` File Walkthrough"
+weight = 2
+sort_by = "weight"
+template = "doc.html"
+insert_anchor_links = "right"
++++
+
+
+Let's go through the type definitions of some of the most used types when working with Graph Store.
+
+## Post
+Here's `sur/post.hoon`
+```
++$  index       (list atom)
++$  uid         [=resource =index]
+::
+::  +sham (half sha-256) hash of +validated-portion
++$  hash  @ux
+::
++$  signature   [p=@ux q=ship r=life]
++$  signatures  (set signature)
++$  post
+  $:  author=ship
+      =index
+      time-sent=time
+      contents=(list content)
+      hash=(unit hash)
+      =signatures
+  ==
+::
++$  indexed-post  [a=atom p=post]
+::
++$  validated-portion
+  $:  parent-hash=(unit hash)
+      author=ship
+      time-sent=time
+      contents=(list content)
+  ==
+::
++$  content
+  $%  [%text text=cord]
+      [%mention =ship]
+      [%url url=cord]
+      [%code expression=cord output=(list tank)]
+      [%reference =uid]
```

Comment:
Out of date

Reply:
Question about the `reference` type: if there is an unmanaged graph (i.e. one that is not tied to a group iirc), then how do references work for that? Cause a `%graph` reference needs a rid to a group. Do references only work for managed graphs?

Reply:
Even unmanaged graphs have a group associated (but it's an unmanaged / hidden group). The group in that case has the same resource id as the graph.